src/rules/incognia-authentication.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * @title Incognia Authentication Rule
+ * @overview Verify if the device logging in is at a trusted location.
+ * @gallery true
+ * @category marketplace
+ *
+ */
+async function incogniaAuthenticationRule(user, context, callback) {
+  const _ = require('lodash@4.17.19')
+
+  const { IncogniaAPI } = require('@incognia/api@1.0.0')
+  const { Auth0UserUpdateUtilities } = require('@auth0/rule-utilities@0.2.0')
+
+  const { INCOGNIA_CLIENT_ID, INCOGNIA_CLIENT_SECRET } = configuration;
+
+  if (!INCOGNIA_CLIENT_ID || !INCOGNIA_CLIENT_SECRET) {
+    console.log('Missing required configuration. Skipping.');
+    return callback(null, user, context);
+  }
+
+  // For this rule to be used, please set 'incognia_authentication_rule' as 'enabled' in your Auth0
+  // application metadata. This can be done in the advanced settings of your app.
+  const incogniaAuthenticationRule = _.get(context, 'clientMetadata.incognia_authentication_rule');
+  if (!incogniaAuthenticationRule || incogniaAuthenticationRule !== 'enabled') {
+    console.log('Incognia authentication rule is not enabled for this client. Skipping');
+    return callback(null, user, context);
+  }
+
+  const installationId = _.get(context, 'request.query.installation_id');
```

Comment:
Could `installation_id` be name-spaced somehow, like `incognia_installation_id`?

Reply:
Sure! We can make that work. I'll update the pull request and also Incognia docs so that `incognia_installation_id` is given in the client during login.